reframe/utility/os_ext.py

```diff
@@ -305,3 +305,25 @@ def git_repo_exists(url, timeout=5):
         return False
     else:
         return True
+
+
+def expandvars(path):
+    """Expand environment variables in ``path`` and
+        perform any command substitution
+
+    This function is the same as ``os.path.expandvars()``, except that it
+    understands also the syntax: $(cmd)`` or `cmd`.
+    """
+    cmd_subst = re.compile(r'(`|\$\()(.*)(`|\))')
```

Comment:
Replace regex with 
```
`.*`|\$\(.*\)
```
Since now matches like:
``` 
`my_command) are allowed
```

Reply:
The problem with your suggestion as the proper regex is that you cannot perform nicely the grouping. You will need to have two groups and sometimes you need to extract group 1, some others group2. I will continue using mine and check explicitly the prefix and suffix.

Reply:
Yes, you can do:
```
cmd_subst = re.compile(r'`(.*)`|\$\((.*)\)')
```
And then use:
```
cmd = cmd_subst_m.groups()[0] or cmd_subst_m.groups()[1]
```